### DISCUSSION TITLE
Incompatibility between sticky scroll and documentSymbol?

### DISCUSSION THREAD

**Discussion by nachocab** [microsoft/vscode-discussions#914] [Extension Development QnA] [answered]
I have a [custom language extension][1] and I've added `folding: {offSide: true}` to my [language-configuration.json][2]. It appears to work, but only for a second: 

[![enter image description here][3]][3]

As soon as the outline panel is loaded, the sticky scroll disappears. 

I think this happens because I'm responding to the `textDocument/documentSymbol` request in [extension.ts][4] incorrectly (these are [the only docs I found][5]):

```js
symbols.push(
  new vscode.DocumentSymbol(
    name,
    "Heading",
    vscode.SymbolKind.Field,
    line.range,
    line.range
  )
);
```

I would appreciate any help.

  [1]: https://github.com/nachocab/vscode-nacho-language/tree/master
  [2]: https://github.com/nachocab/vscode-nacho-language/blob/master/language-configuration.json#L42
  [3]: https://i.stack.imgur.com/Ojeig.gif
  [4]: https://github.com/nachocab/vscode-nacho-language/blob/master/src/extension.ts#L29
  [5]: https://code.visualstudio.com/api/language-extensions/programmatic-language-features#show-all-symbol-definitions-within-a-document

**Comment by isidorn** [comment]
@aiday-mar might be able to help here.

**Comment by aiday-mar** [comment]
Hey thank you for filing this issue. I am trying to reproduce this issue. What is the file extension you use for the nacho language?

**Comment by nachocab** [comment]
@aiday-mar I use .txt with markdown h2-h6 indented headers:

```text
## Level 2
    ### Level 3
          # a comment
           Some content

## Another Level 2
    Some other comment
```

**Reply by aiday-mar** [reply]
I see so will I be able to reproduce this issue if I install the nacho extension from VS Code, or is there a modified version of it I am to use? In the latter case, could you send the source code you are working on

**Reply by nachocab** [reply]
Yeah, just install the extension and use 'Change Language Mode' to set a file to 'nacho'. Then write something like what I sent you and add newlines to see the sticky scroll. They show up briefly when you reload the window.

**Comment by nachocab** [comment, marked as answer]
@aiday-mar I think the problem is that the `range` in the `DocumentSymbol`s has the same `startLine` and `endLine`. If I do a proper AST and I calculate the right `endLine`s, I don't run into this issue.

But still, now that vscode allows folding through indentation, it should be automatically calculated.

**Reply by aiday-mar** [reply]
I see, I am glad to see that you have figured out the issue. Thank you for the feedback. Since the issue appears to be resolved, we can close it.

**Reply by nachocab** [reply]
I'm about to publish version 0.9.0 of my extension, which fixes the issue. But feel free to use v0.8.5 for testing purposes.
